lib/gears/surface.lua

```diff
@@ -45,15 +45,28 @@ function surface.load_uncached_silently(_surface, default)
     if cairo.Surface:is_type_of(_surface) then
         return _surface
     end
+
+
     -- Strings are assumed to be file names and get loaded
     if type(_surface) == "string" then
-        local err
         file = _surface
+
+        local err
+
+        -- `s` is an undocumented workaround until #908 is implemented
         _surface, err = capi.awesome.load_image(file)
+
         if not _surface then
             return get_default(default), err
         end
+
+        -- The shims implement load_image() to return a surface directly,
+        -- instead of a lightuserdatum.
+        if cairo.Surface:is_type_of(_surface) then
+            return _surface
+        end
```

Comment:
Please take a look at the third commit and remove all changes except for this one (especially the addition of a wrong comment).

Reply:
This comment still applies but now to the second commit and in a different form: It only adds empty lines.